go-controller/pkg/ovn/controller/services/lb_config.go

```diff
@@ -648,56 +673,57 @@ func buildPerNodeLBs(service *corev1.Service, configs []lbConfig, nodes []nodeIn
 
 				for _, vip := range vips {
 					isv6 := utilnet.IsIPv6String((vip))
-					// build switch rules
-					targets := switchV4targets
-					if isv6 {
-						targets = switchV6targets
-					}
 
 					if cfg.externalTrafficLocal && cfg.hasNodePort {
 						// add special masqueradeIP as a vip if its nodePort svc with ETP=local
 						mvip := config.Gateway.MasqueradeIPs.V4HostETPLocalMasqueradeIP.String()
-						targetsETP := joinHostsPort(switchV4TargetIPs, cfg.clusterEndpoints.Port)
+						targetsETP := nodeSwitchV4TargetAddrs
 						if isv6 {
 							mvip = config.Gateway.MasqueradeIPs.V6HostETPLocalMasqueradeIP.String()
-							targetsETP = joinHostsPort(switchV6TargetIPs, cfg.clusterEndpoints.Port)
+							targetsETP = nodeSwitchV6TargetAddrs
 						}
 						switchRules = append(switchRules, LBRule{
 							Source:  Addr{IP: mvip, Port: cfg.inport},
 							Targets: targetsETP,
 						})
 					}
+
 					if cfg.internalTrafficLocal && util.IsClusterIP(vip) { // ITP only applicable to CIP
-						targetsITP := joinHostsPort(switchV4TargetIPs, cfg.clusterEndpoints.Port)
+						targetsITP := nodeSwitchV4TargetAddrs
 						if isv6 {
-							targetsITP = joinHostsPort(switchV6TargetIPs, cfg.clusterEndpoints.Port)
+							targetsITP = nodeSwitchV6TargetAddrs
 						}
 						switchRules = append(switchRules, LBRule{
 							Source:  Addr{IP: vip, Port: cfg.inport},
 							Targets: targetsITP,
 						})
 					} else {
+						// build switch rules
```

Comment:
Note: moved this down here from line 651 because `targets` is only used here and it wasn't super evident just by briefly looking at the code